provision/acc_provision/acc_provision.py

```diff
@@ -1365,6 +1365,16 @@ def check_overlapping_subnets(config):
     return True
 
 
+def check_image_pull_secret(config):
+    # Check if the image_pull_secret is valid
+    image_pull_secret = config["registry"]["image_pull_secret"]
+    # This is the regex used by kubectl to validate objects names
+    pattern = "[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*"
```

Comment:
Try to compare this with other examples that people might have posted for validating DNS subdomain names, and accordingly lets add a UT which validates, both, the positive and the negative cases.

Reply:
okay let me understand how to write a UT for validating DNS subdomain names then may be similarly try to mimic the work for our use case - image_pull_secret 

I guess the understanding is to add UT here - https://github.com/noironetworks/acc-provision/blob/master/provision/acc_provision/test_main.py
where 67 unit tests are already written for other work.
@snaiksat

Reply:
To clarify, what I meant is compare the regex pattern we have used with the one others have used for DNS subdomain names to ensure we are not missing anything.

Reply:
okay makes sense, thanks for clarifying! @snaiksat 

Regarding adding UT 
(67 existing ones + adding 1 new one for image_pull_secret) 
I guess with respect to this the understanding is right from my end?